### ISSUE TITLE
Browse query does not iterate properly over all items

### ISSUE THREAD

**Issue by PaulinSanselme-Concord** [algolia/algoliasearch-client-java#341]
Hi,

When browsing results from Algolia, only the first page of results are retrieved.
```
final Index<MyClass> index = client.initIndex(indexName, MyClass.class);
final Query query = new Query()
        .setFilters("")
        .setAttributesToRetrieve(Arrays.asList("objectID"))
        .setAttributesToHighlight(Collections.emptyList())
        .setAttributesToSnippet(Collections.emptyList())
        .setHitsPerPage(1000)
        .setDistinct(Distinct.of(false));

int count = 0;
for (final MyClass item : index.browse(query)) {
	++count;
}
System.out.println(count); // 1000 even if more records should be retrieved.
```

The issue comes from the method `hasNext` from `IndexIterator` which should look for others records when `currentIterator` is consumed.

As a workaround, I tried to execute the following code :
```
final Index<MyClass> index; // same as previous
final Query query; // same as previous

String cursor = null;
IndexIterable<MyClass> items = index.browseFrom(query, cursor);

while (items.iterator().hasNext()) {
	for (final MyClass item : items) {
		++count;
	}
	cursor = items.getCursor();
	items = index.browseFrom(query, cursor);
}
System.out.println(count); // Never reach this line.
```
But unfortunately, even if I iterate over all records and not only the 1000 first ones, it seems that there is a kind of cycle which lead me to iterate endlessly over all the records.

**Comment by ElPicador** [comment]
Yes you are right. The test on browse was bad. 
I've updated the test and it fails: https://github.com/algolia/algoliasearch-client-java-2/pull/342

I'll work on a fix ASAP.

And thanks for the report and all the details :)

**Comment by ElPicador** [comment]
I've released version 2.11.1 that fixes the issue.

Could you try it?

**Comment by PaulinSanselme-Concord** [comment]
Yes it solves the issue
